mounch.py

```diff
@@ -130,19 +130,16 @@ def main():
         cached_entries[chosen_id] = 0
     else:
         cached_entries[chosen_id] += 1
-    cache_file.write_text("\n".join(
-        [f"{entry} {cached_entries[entry]}" for entry in cached_entries]))
+    cache_file.write_text('\n'.join(
+        [f'{entry} {freq}' for entry, freq in cached_entries.items()]))
 
     binarypath = pathlib.Path(chosen['binary']).expanduser()
-    if binarypath.exists():
-        binary = binarypath
-    else:
-        binary = shutil.which(chosen['binary'])
-        if not binary:
-            binary = pathlib.Path(f"~/bin/desktop/{chosen['binary']}").expanduser()
-            if not binary.exists():
-                print(f"Cannot find {chosen['binary']}")
-                sys.exit(1)
+    binary = shutil.which(binarypath)
+    if not binary:
+        binary = pathlib.Path(f"~/bin/desktop/{chosen['binary']}").expanduser()
```

Comment:
I think we can drop the looking into ~/bin/desktop/ which is something i had just because i wasn't loading my user path properly in desktop

Reply:
I agree, as `shutil.which` looks up for an executable in $PATH and supports absolute paths but I didn't want to interfere with your way of doing things.

Reply:
Did it.

Reply:
awesome thank you so much!